include/manif/impl/eigen.h

```diff
@@ -134,9 +134,8 @@ struct traitscast<Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxC
  *             | x  0 |
  */
 template <typename _Scalar>
-typename std::enable_if<std::is_arithmetic<_Scalar>::value,
-                        Eigen::Matrix<_Scalar, 2, 2>>::type
-skew(const _Scalar v)
+Eigen::Matrix<_Scalar, 2, 2>
+skew2x2(const _Scalar v)
```

Comment:
Can you please provide an example which raises the issue here? I'd prefer to find a better suited sfinae condition rather than renaming the function. In case such condition doesn't exists, or is too wobbly, I would rename the function `skew2` and the other one `skew3` to better stick with `Eigen` naming.

Reply:
Here's an example:

```
#include <manif/impl/traits.h>
#include <manif/impl/eigen.h>

// example "autodiff" type
struct MyDouble
{
  MyDouble(double d)
    : d_(d)
  {}

  double d_;
};

int main()
{
  Eigen::Matrix<double, 6, 1> v = Eigen::Matrix<double, 6, 1>::Zero();

  // these should call the scalar version
  manif::skew(1.0);
  manif::skew(MyDouble(1));

  // these should call the vector version
  manif::skew(Eigen::Vector3d{1, 2, 3});
  manif::skew(v.head<3>());
}
```
It seems like the scalar version should be called for all types that are not Eigen types, but I couldn't find a good condition to express that something is not an Eigen type. I tried ```!is_base_of<MatrixBase<T>, T>``` but this does not work since ```VectorBlock``` does not inherit from ```MatrixBase```.

Let me know if you find a working condition, otherwise I'll go ahead with the renaming.

Reply:
Thanks for the small example, I'll have a look at what's possible to do.  
At the end of the day we can always have a custom traits in `manif` that's specialized for each AD scalar type, something along the line:
```cpp
template <typename Scalar> struct is_ad : std::integral_constant<bool, false> { };
...
template <> struct is_ad<MyDouble> : std::integral_constant<bool, true> { };
...
template <typename Scalar>
typename std::enable_if<std::is_arithmetic<Scalar>::value || is_ad<Scalar>::value, Matrix>
skew( ... ) { ... }
```